specificatie/opschortingbijhouding.yaml

```diff
@@ -0,0 +1,43 @@
+openapi: 3.0.3
+info:
+  title: OpschortingBijhouding definities
+  version: 2.0.0
+  contact: {}
+paths: {}
+components:
+  schemas:
+    OpschortingBijhoudingBasis:
+      type: object
+      properties:
+        reden:
+          $ref: 'https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.2.0/api-specificatie/common.yaml#/components/schemas/Waardetabel'
+    AbstractOpschortingBijhouding:
+      type: object
+      description: |
+          * **datum**: de datum waarop de bijhouding van de persoonsgegevens is gestaakt.
+      required:
+        - type
+      properties:
+        type:
+          type: string
+        reden:
+          $ref: 'https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.2.0/api-specificatie/common.yaml#/components/schemas/Waardetabel'
+      discriminator:
+        propertyName: type
+        mapping:
+          GbaOpschortingBijhouding: '#/components/schemas/GbaOpschortingBijhouding'
+          OpschortingBijhouding: '#/components/schemas/OpschortingBijhouding'
+    OpschortingBijhouding:
+      allOf:
+        - $ref: '#/components/schemas/AbstractOpschortingBijhouding'
+        - type: object
+          properties:
+            datum:
+              $ref: 'datum.yaml#/components/schemas/AbstractDatum'
+    GbaOpschortingBijhouding:
+      allOf:
+        - $ref: '#/components/schemas/AbstractOpschortingBijhouding'
+        - type: object
+          properties:
+            datum:
+              $ref: 'datum.yaml#/components/schemas/GbaDatum'
```

Comment:
De componenten 'OpschortingBijhouding' en 'GbaOpschortingBijhouding' hebben beide een 'datum' property. In de eerste verwijst deze naar het component 'AbstractDatum' waardoor je in feite de mogelijkheid hebt om uit 5 verschillende datum types te kiezen. Bij de tweede verwijst deze naar het component 'GbaDatum' waarmee je zou verwachten dat je alleen over het GbaDatum type zou kunnen beschikken. Binnen Redoc zie ik echter dat je ook in dit geval de 5 keuzes krijgt. Wellicht is het een bug in Redoc, wellicht een bewuste implementatie maar het lijkt me niet wat we willen.

In mijn eigen fork heb ik daarom het component 'GbaDatum' aangepast. Ik heb een component 'GbaDatumBeperkt' aangemaakt met de volgende vulling:

    GbaDatumBeperkt:
      description: Datum conform LO GBA
      type: string
      pattern: ^[0-9]{8}$
      example: "20210801"

'GbaDatum' heb ik gewijzigd in:

    GbaDatum:
      allOf:
        - $ref: '#/components/schemas/AbstractDatum'
        - type: object
          properties:
            datum:
              $ref: '#/components/schemas/GbaDatumBeperkt'

En de $ref in het component 'GbaOpschortingBijhouding' verwijst nu naar 'GbaDatumBeperkt'.

Zie de gevolgen daarvan in https://melsk-r.github.io/Haal-Centraal-BRP-bevragen/redoc-io#operation/GetPersoon.

Reply:
Hetzelfde speelt volgens mij overigens op meerdere plaatsen. Ook in het component 'GbaInOnderzoek' moet de $ref n.m.m. verwijzen naar 'GbaDatumBeperkt'.

Reply:
Wanneer het enige verschil tussen de GBA variant en de HC variant is dat in de GBA variant alleen een GBA datum kan worden geleverd, is polyforme denk ik niet nuttig. Enig nu is voor de implementator van de GBA-variant en implementator van de proxy duidelijk specificeren. Daarvoor hoeven we de specificaties voor alle gebruikers niet complexer te maken.

Reply:
@melsk-r, het is niet een bug in Redoc (denk ik), maar meer hoe inheritance werkt. Omdat van AbstractDatum wordt geërfd krijgt je niet alleen het type mee, maar ook de descriminator en mapping. Het is in dit geval dus beter om een ander GbaDatum type (en niet een string zoals je het nu hebt gedaan, want hierdoor verschilt jouw datum (type: string) van de andere type datums (type: object)) te introduceren.
@fsamwel, ik wacht nog op reactie op de vraag of OpschortingBijhouding nog andere properties heeft dan reden en datum. Zo niet, dan geldt hier hetzelfde als bij Naam en is de GBA variant helemaal niet nodig